src/Mapping/Column.php

```diff
@@ -24,13 +26,25 @@ public function __construct(
         public readonly int|null $precision = null,
         public readonly int|null $scale = null,
         public readonly bool $unique = false,
-        public readonly bool $nullable = false,
+        private bool|null $nullable = null,
         public readonly bool $insertable = true,
         public readonly bool $updatable = true,
         public readonly string|null $enumType = null,
         public readonly array $options = [],
         public readonly string|null $columnDefinition = null,
         public readonly string|null $generated = null,
     ) {
+        $this->nullableSet = $nullable !== null;
+        $this->nullable  ??= false;
+    }
+
+    public function isNullable(): bool
+    {
+        return $this->nullable;
+    }
+
+    public function isNullableSet(): bool
+    {
+        return $this->nullableSet;
     }
```

Comment:
Why do we use getter methods instead of public properties now?

Reply:
I added isNullable() because of type safety, and nullableSet so it cannot be changed later.. I removed those getters now, it's solved better I think (I separated constructor property), and without BC break.